app/src/main/java/io/gnosis/safe/ui/StartActivity.kt

```diff
@@ -58,6 +58,12 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler {
         setupNav()
 
         handleNotifications(intent)
+
+        if (settingsHandler.appStartCount <= 1) {
+            Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.createPasscodeFragment, Bundle().apply {
```

Comment:
on first app start user would have to accept terms and conditions. Could we extend flow on a Terms screen, so that after clicking accept user will be asked to setup a passcode?

Reply:
on the other thought, we have deal with already installed apps and in case of upgrades this would works better. Could you move it to onResume and extend the when statement there?

Reply:
Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate... should be moved to a private method (something like setupPasscode())

Reply:
Moving this into onResume has no advantage. We only want to do this once after the first app start.  And the disadvantage that the check not to do it will be more complicated during the first use of the app.

Reply:
I extracted the navigation into a function.

Reply:
The order is:
1st app start:
- accept terms
- update software
- setup passcode

Reply:
I think it makes sense to check this order, when the last PR is being merged.